backend/stakepoold/server.go

```diff
@@ -242,18 +324,118 @@ func runMain() int {
 	log.Infof("Connected to dcrd (JSON-RPC API v%s) on %v",
 		nodeVer.String(), curnet.String())
 
-	_, height, err := nodeConn.GetBestBlock()
+	curHash, curBlock, err := nodeConn.GetBestBlock()
 	if err != nil {
 		log.Errorf("unable to get bestblock from dcrd: %v", err)
 		return 8
 	}
-	ctx.blockheight = height
+
+	// prune save data
+	err = pruneData(ctx)
+	if err != nil {
+		log.Warnf("pruneData error: %v", err)
+	}
+
+	// load liveTickets
+	err = loadData(ctx, "LiveTickets")
+	if err != nil {
+		log.Warnf("unable to load LiveTickets from disk cache: %v", err)
+	} else {
+		log.Infof("Loaded %v LiveTickets from disk cache", len(ctx.liveTicketsMSA))
+	}
+
+	// load ManuallyAddedTickets
+	// TODO implement in dcrstakepool first
+
+	// load stats
+	err = loadData(ctx, "Stats")
+	if err != nil {
+		log.Warnf("unable to load Stats from disk cache: %v", err)
+	} else {
+		log.Info("Loaded Stats from disk cache")
+	}
+
+	// load userVotingConfig from disk cache if necessary
+	if len(userVotingConfig) == 0 {
+		err = loadData(ctx, "UserVotingConfig")
+		if err != nil {
+			// XXX we could possibly die out here but it's probably better
+			// to let stakepoold vote with default preferences rather than
+			// not vote at all
+			log.Warnf("unable to load user voting preferences from disk "+
+				"cache: %v", err)
+		} else {
+			log.Infof("Loaded UserVotingConfig for %d users from disk cache",
+				len(userVotingConfig))
+		}
+	}
+
+	statsBlockHeight := atomic.LoadInt64(&ctx.stats.BlockHeight)
+	atomic.StoreInt64(&ctx.lastVotedBlock, statsBlockHeight)
+	log.Infof("updated lastVotedBlock to %v", statsBlockHeight)
+
+	ctx.printStats()
+
+	deferGetTickets := false
+
+	// get tickets now if we don't have any
+	if len(ctx.liveTicketsMSA) == 0 {
+		timenow := time.Now()
+		ctx.mtx.Lock()
+		ctx.liveTicketsMSA = walletGetTickets(ctx)
```

Comment:
Right, if we're stuck doing GetTickets() because we have 0 tickets and a block comes in, we will miss it.

This is why I added caching -- so stakepoold can start immediately voting the tickets it loads from disk.  Then when gettickets finishes processing, it will delete/append the differences.